generation/GenerateTables.c

```diff
@@ -0,0 +1,177 @@
+
+    /* Copy backward.  */
+    for (int rev = BLOCK_SIZE - 1; rev > 0; --rev)
+      block->data[rev] = block->data[rev - 1];
+
+    block->data[0] = tmp;
+  }
+}
+
+/* Work-horse for generate_mul_tbl.  */
+static unsigned char poly_mul(unsigned char left, unsigned char right) {
+  /* p(x) = x**8 + x**7 + x**6 + x + 1:
+       0b111000011 = 0x1C3
+
+     with no MSB: 0xC3.  */
+  unsigned char ret = 0;
+  while (left && right) {
+    if (right & 1)
+      ret ^= left;
+
+    left = (left << 1) ^ (left & 0x80 ? 0xC3 : 0x00);
+    right >>= 1;
+  }
+
+  return ret;
+}
+
+/* Pregenerations.  */
+void generate_mul_tbl(unsigned char Muls[][16]) {
+  for (int i = 0; i < 256; ++i)
+    for (int j = 0; j < 16; ++j)
+      Muls[i][j] = poly_mul(i, Lin[j]);
+}
+
+void generate_coef_tbl(Block Coefs[][BIT_IN_BYTES], unsigned char Muls[][16]) {
+  for (int i = 0; i < NUM_ROUNDS / 2 - 1; ++i)
+    for (int j = 0; j < BIT_IN_BYTES; ++j) {
+      Block *c = &Coefs[i][j];
+      memset(c, 0, BLOCK_SIZE);
+      c->data[BLOCK_SIZE - 1] = i * 8 + j + 1;
+      apply_l(c, Muls);
+    }
+}
+
+void generate_LS_tbl(unsigned char LS_tbl[][256][16]) {
+#define MATR_POWER 4
+  unsigned char l_tbl[BLOCK_SIZE][BLOCK_SIZE] = {0};
+
+  /* Init matrix.  */
+  for (int i = 0; i < BLOCK_SIZE; ++i)
+    for (int j = 0; j < BLOCK_SIZE; ++j)
+      if (i - 1 == j)
+        l_tbl[i][j] = 1;
+      else if (i == 0)
+        l_tbl[i][j] = Lin[j];
+
+  /* Powering the matrix.  */
+  unsigned char tmp[BLOCK_SIZE][BLOCK_SIZE] = {0};
+  for (int p = 0; p < MATR_POWER; ++p) {
+    for (int i = 0; i < BLOCK_SIZE; ++i)
+      for (int j = 0; j < BLOCK_SIZE; ++j)
+        for (int k = 0; k < BLOCK_SIZE; ++k)
+          tmp[i][j] ^= poly_mul(l_tbl[i][k], l_tbl[k][j]);
+
+    memcpy(l_tbl, tmp, 256 * sizeof(unsigned char));
+    memset(tmp, 0, 256 * sizeof(unsigned char));
+  }
+
+  for (int i = 0; i < BLOCK_SIZE; ++i)
+    for (int j = 0; j < 256; ++j)
+      for (int k = 0; k < BLOCK_SIZE; ++k)
+        LS_tbl[i][j][k] ^= poly_mul(S[j], l_tbl[k][i]);
+}
+
+int main(int argc, char* argv[]) {
+    const char* coefs_file_path = "./coefs.dat";
+    const char* muls_file_path = "./muls.dat";
+    const char* ls_tbl_file_path = "./ls_tbl.dat";
+
+    Block Coefs[NUM_ROUNDS / 2 - 1][BIT_IN_BYTES];
+    unsigned char Muls[256][16];
+    unsigned char LS_tbl[16][256][16];
+
+    generate_mul_tbl(Muls);
+    generate_coef_tbl(Coefs, Muls);
+    generate_LS_tbl(LS_tbl);
+
+    FILE* fd1 = fopen(muls_file_path, "a");
+    if (fd1) {
+      fprintf(fd1, "static unsigned Muls[256][16] = {");
+      for (size_t i = 0; i < 256; ++i) {
+          for (size_t j = 0; j < 16; ++j) {
+            if (i == 255 && j == 15) {
+              fprintf(fd1, "0x%x}", Muls[i][j]);
+            } else {
+              fprintf(fd1, "0x%x, ", Muls[i][j]);
+            }
+          }
+          fprintf(fd1, "\n");
+      }
+      fclose(fd1);
+    } else {
+      perror("Error open file\n");
```

Comment:
Remove '\n' from perror. Perror output looks like this:
```
Error open file: error_message_here
```

But now you have '\n' and getting:
```
Error open file:
error_message_here
```